test/integration/main_test.go

```diff
@@ -349,7 +378,7 @@ func testUploadFile(ctx context.Context, tb testing.TB, bucket, object string, d
 	o := gcsClient.Bucket(bucket).Object(object)
 
 	// For an object that does not yet exist, set the DoesNotExist precondition.
-	o = o.If(storage.Conditions{DoesNotExist: true})
+	// o = o.If(storage.Conditions{DoesNotExist: true})
```

Comment:
Why do you comment this?

Reply:
https://github.com/googleapis/google-cloud-go/blob/main/storage/storage.go#L1724-L1727
Because the upload container to steps, first is write the object, second is add metadata, if DoseNotExist is set to true, the add metadata will fail because the this object is already there.

If we can to keep the conditions, then we would need a new writer to add metadata, I think that's not necessary.

Reply:
Found a new way to write metadata which allows us to keep using preconditions.